launcher/modplatform/modrinth/ModrinthPackManifest.cpp

```diff
@@ -95,19 +97,15 @@ void loadIndexedVersions(Modpack& pack, QJsonDocument& doc)
 
 auto validateDownloadUrl(QUrl url) -> bool
 {
+    static QSet<QString> domainWhitelist{
+        "cdn.modrinth.com",
+        "github.com",
+        "raw.githubusercontent.com",
+        "gitlab.com"
+    };
+
     auto domain = url.host();
-    if(domain == "cdn.modrinth.com")
-        return true;
-    if(domain == "edge.forgecdn.net")
-        return true;
-    if(domain == "media.forgecdn.net")
-        return true;
-    if(domain == "github.com")
-        return true;
-    if(domain == "raw.githubusercontent.com")
-        return true;
-
-    return false;
+    return domainWhitelist.contains(domain);
 }
```

Comment:
Forgive me for my ignorance (I don't know anything about the Modrinth modpack format), but why does the launcher need to check for approved download URLs? If Modrinth allows (or bans) certain domains, doesn't that mean that modpacks downloaded from Modrinth will already only include allowed domains, and thus checking again here is redundant and adds extra maintenance?

Reply:
Well, it's in the specification so we implemented it. It will be checked by Modrinth themselves, that's right, but `.mrpack`s need not necessarily come from Modrinth, since it's an open format. So, this is just so that we're aligned with Modrinth on what should be allowed.

As of my latest commits, there'll simply be a warning if there's an URL that is not whitelisted, instead of failing, so it prevents things like malicious URLs being thrown at the user without their knowledge, when the pack comes from a third-party.

(Also, maybe you should get into the Discord, we discuss this kinds of things in there x) )

Reply:
Ah, that makes sense.

> (Also, maybe you should get into the Discord, we discuss this kinds of things in there x) )

Ah, sorry for the duplicated answer then. Maybe in the future :) (I don't have a Discord account I want to associate with my GH stuff at the moment :p)

Reply:
> Maybe in the future :)

I'll wait for it then :)
(you can also use the matrix bridge if you feel like it ;))